src/providers/proxy/proxy_id.c

```diff
@@ -1488,6 +1488,152 @@ static int get_initgr(TALLOC_CTX *mem_ctx,
     return ret;
 }
 
+static int compare_group_members(struct proxy_id_ctx *ctx,
+                       TALLOC_CTX *tmp_ctx,
+                       struct sss_domain_info *dom,
+                       long int num_gids,
+                       gid_t *gids,
+                       char *db_groupname,
+                       bool *group_found)
+{
+    int j = 0;
+    int ret = EOK;
+    size_t buflen = 0;
+    char *buffer = NULL;
+    char *network_groupname = NULL;
+    enum nss_status status;
+    struct group *grp = NULL;
+    bool delete_group = false;
+
+    grp = talloc(tmp_ctx, struct group);
+    if (!grp) {
+        ret = ENOMEM;
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc() failed\n");
+        goto done;
+    }
+
+    // group 0 is the primary group so it can be skipped
+    for (j = 1; j < num_gids; j++) {
+        do {
+            /* always zero out the grp structure */
+            memset(grp, 0, sizeof(struct group));
+            buffer = grow_group_buffer(tmp_ctx, &buffer, &buflen);
+            if (!buffer) {
+                ret = ENOMEM;
+                goto done;
+            }
+
+            status = ctx->ops.getgrgid_r(gids[j], grp, buffer, buflen, &ret);
+
+            ret = handle_getgr_result(status, grp, dom, &delete_group);
```

Comment:
We don't actually use `delete_group`. Perhaps it would make sense to `handle_getgr_result()` to support `NULL` as this arg...